### DISCUSSION TITLE
React/JS - Spinning Wheel

### DISCUSSION THREAD

**Discussion by rubenmarcus** [frontendbr/forum#1288] [Perguntas e Respostas]
Olá pessoal tudo bem? Estou fazendo um site pra um cliente em React, e tenho esse componente, onde o usuário pode clicar no circulo maior e arrastar ( no caso rodar ) a wheel, a vontade. Estou tendo uma grande dor de cabeça, porque não faço idéia de como fazer isso. O designer, já aceitou o efeito apenas por duas setas ( eu pego o click do usuario e troco a classe do componente pai, rotacionando todos os elementos filhos com transform rotate e mudando a position dos elementos (dou rotate no circulo, e os elementos, as labels pai, e as labels filhos com icone eu so mudo a posição) ) , mas ainda sim o cliente e o designer gostaria de ver o usuário rotacionando a seu bel prazer com o dedo (pensando em mobile)

Não tenho idéia de como começar isso. Nem como procurar no google. Já procurei por react touch, react rotate ( achei umas libs boas de 3d), react carousel rotate, js wheel rotate, mas nada.

Se alguem souber e poder dar uma luz, ficarei grato.

![](https://i.imgur.com/9LyMX9Q.png)

**Comment by bizarro** [comment]
Esse tipo de coisa é aquelas que você precisa ir um pouco além, acho que o [Draggable do TweenMax](https://greensock.com/draggable) resolve seu problema, eu usei num site chamado [Say Yes To The World 2](https://sayyestotheworld.com/global/en) que tem uma wheel que altera o conteúdo também. 

Basicamente é só você adicionar o Draggable numa `div` e usar o rotation do `TweenMax` em outra.

**Comment by samiraguiar** [comment]
Procura por potentiometer ou knob:
https://github.com/hugozap/react-rotary-knob
http://anthonyterrien.com/demo/knob/
https://codepen.io/solipsistacp/pen/ORYyPw/
https://www.cssscript.com/touch-enabled-knob-control-pure-javascript-jim-knopf/
https://www.cssscript.com/tag/knob/

**Comment by rubenmarcus** [comment]
esse do http://anthonyterrien.com/demo/knob/

parece bacana, mas usa jquery, queria evitar, se não tiver como vou usar. Eu vou entregar o site com as setas mesmo, e pedir mais um tempo pra fazer esse knob, porque vai levar mesmo em 1 dia não rola fazer eu acho

**Comment by allansduarte** [comment]
Interessante o desafio, compartilhe com a gente a solução quando resolver
